File: Fase 2/ServidorRevistas/src/main/java/objetos/Revista.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package objetos;

import java.util.ArrayList;
import java.util.Date;


/**
 *
 * @author fernanrod
 */
public class Revista {
    private Integer id;
    private String nombre;
    private String descripcion;
    private Date fechaPublicacion;
    private String categoria;
    private ArrayList<Etiqueta> etiquetas;
    private String estadoSuscripciones;
    
    public Revista(Integer id, String nombre, String descripcion, Date fechaPublicacion, String categoria, ArrayList<Etiqueta> etiquetas, String estadoSuscripciones) {
        this.id = id;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.fechaPublicacion = fechaPublicacion;
        this.categoria = categoria;
        this.etiquetas = etiquetas;
        this.estadoSuscripciones = estadoSuscripciones;
    }
    
    public Integer getId() {
        return this.id;
    }
    public void setId(Integer id) {
        this.id = id;
    }
    public String getNombre() {
        return this.nombre;
    }
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    public String getDescripcion() {
        return this.descripcion;
    }
    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }
    public Date getFechaPublicacion() {
        return this.fechaPublicacion;
    }
    public void setFechaPublicacion(Date fechaPublicacion) {
        this.fechaPublicacion = fechaPublicacion;
    }
    public String getCategoria() {
        return this.categoria;
    }
    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }
    public void setEtiquetas(ArrayList<Etiqueta> etiquetas) {
        this.etiquetas = etiquetas;
    }  
    public ArrayList<Etiqueta> getEtiquetas() {
        return etiquetas;
    }
    public String getEstadoSuscripciones() {
        return this.estadoSuscripciones;
    }
    public void setEstadoSuscripciones(String estadoSuscripciones) {
        this.estadoSuscripciones = estadoSuscripciones;
    }
}

File: Fase 2/revistas-app/src/app/objetos/editor/Revista.ts
import { Categoria } from "../base/Categoria";
import { Etiqueta } from "../base/Etiqueta";

export class Revista {
  private _id: number | null;
  private _nombre: string;
  private _descripcion: string;
  private _fechaPublicacion: Date;
  private _categoria: Categoria;
  private _etiquetas: Array<Etiqueta>;
  private _estadoSuscripciones: string | null;

  constructor(id: number | null, nombre: string, descripcion: string, fechaPublicacion: Date, categoria: Categoria, etiquetas: Array<Etiqueta>, estadoSuscripciones: string|null) {
    this._id = id;
    this._nombre = nombre;
    this._descripcion = descripcion;
    this._fechaPublicacion = fechaPublicacion;
    this._categoria = categoria;
    this._etiquetas = etiquetas;
    this._estadoSuscripciones = estadoSuscripciones;
  }

  public get id(): number | null{
    return this._id;
  }

  public get nombre(): string{
    return this._nombre;
  }

  public get descripcion(): string{
    return this._descripcion;
  }


  public get fechaPublicacion(): Date{
    return this._fechaPublicacion;
  }

  public get categoria(): Categoria{
    return this._categoria;
  }

  public get etiquetas(): Array<Etiqueta> {
    return this._etiquetas;
  }

  public get estadoSuscripciones(): string|null {
    return this._estadoSuscripciones;
  }

  public set estadoSuscripciones(value: string | null) {
    this._estadoSuscripciones = value;
  }
}
